app-modules/knowledge-base/database/migrations/2024_10_29_183334_data_add_remainig_slug_in_knowledge_base_categories_table.php

```
@@ -0,0 +1,81 @@
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Database\QueryException;
+use Illuminate\Database\Migrations\Migration;
+use AidingApp\KnowledgeBase\Models\KnowledgeBaseCategory;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        $categories = KnowledgeBaseCategory::whereNull('slug')->get();
+
+        foreach ($categories as $category) {
+            $baseSlug = Str::slug($category->name);
+            $slug = $baseSlug;
+            $attempts = 0;
+
+            while ($attempts < 15) {
+                try {
+                    DB::beginTransaction();
+
+                    $category->slug = $slug;
+                    $category->save();
+
+                    DB::commit();
+
+                    break;
+                } catch (QueryException $e) {
+                    DB::rollBack();
+
+                    if ($e->errorInfo[1] === 7) {
+                        $attempts++;
+                        $slug = $baseSlug . '-' . $attempts;
+                    } else {
+                        throw $e;
+                    }
+                }
+            }
+
+            if ($attempts == 15) {
+                Log::info("Failed to generate a unique slug after 15 attempts for category ID {$category->name}\n");
```

Let's throw an exception here, this will need to totally fail if there is a problem.